Moq.AutoMock/Resolvers/MockResolver.cs

```diff
@@ -38,29 +38,24 @@ public void Resolve(MockResolutionContext context)
 
             var mockType = typeof(Mock<>).MakeGenericType(context.RequestType);
 
-            bool mayHaveDependencies = context.RequestType.IsClass 
+            bool mayHaveDependencies = context.RequestType.IsClass
                                        && !typeof(Delegate).IsAssignableFrom(context.RequestType);
-            
-            object?[] constructorArgs;
+
+            //Invoke the Mock<T>(MockBehavior, params object[]) constructor
+            object?[] constructorArgs = new object?[2];
+            constructorArgs[0] = _mockBehavior;
+
             if (mayHaveDependencies)
             {
-                constructorArgs = context.AutoMocker.CreateArguments(context.RequestType);
+                constructorArgs[1] = context.AutoMocker.CreateArguments(context.RequestType);
```

Comment:
Does this work? I'm trying to replicate a `params` call inside `Activator.CreateInstance`. I believe this resolves to:

```
Activator.CreateInstance(Type, object[])
```

which I believe triggers

```
new Mock<Type>(new object[]{ _mockBehavior, context.AutoMocker.CreateArguments() })
```

I understand that it compiles. But does it functionally work?

Reply:
Yes, all of the test pass with it. The key here is that methods with `params` parameter, compile down to a signature that does without (albeit the parameter contains a compiler generated attribute). The other side of the coin is invoking a method that has a [`params` parameter](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/params) if what you pass is an array of the appropriate type, the compile will not wrap the array in another array and instead just pass it strait through.

Small sample showing it:
```
class Program
{
    static void Main()
    {
        object[] args = new object[2];
        args[0] = typeof(int);
        args[1] = new object[] { 1, 2 };
        Activator.CreateInstance(typeof(NewMe), args);
    }
}

class NewMe
{
    public NewMe(Type type, params object[] values)
    {
        Console.WriteLine("First");
    }

    public NewMe(object[] values)
    {
        Console.WriteLine("Second");
    }
}
```
Outputs
```
> First
```